pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1296,6 +1296,20 @@ func (r *ReconcileClusterDeployment) ensureClusterDeprovisioned(cd *hivev1.Clust
 		cdLog.Info("PreserveOnDelete=true but creating deprovisioning request as cluster was never successfully provisioned")
 	}
 
+	// Stop waiting for deprovision if the abandon-deprovision annotation is true
+	if value, ok := cd.Annotations[constants.AbandonDeprovisionAnnotation]; ok {
+		logger := cdLog.WithField(constants.AbandonDeprovisionAnnotation, value)
+		if abandon, err := strconv.ParseBool(value); abandon && err == nil {
```

Comment:
```suggestion
		if abandon, err := strconv.ParseBool(value); err == nil && abandon {
```
wouldn't that be more appropriate?